pkg/kv/kvserver/replica_proposal_buf.go

```diff
@@ -899,7 +920,18 @@ func (b *propBuf) forwardClosedTimestampLocked(closedTS hlc.Timestamp) bool {
 	return b.assignedClosedTimestamp.Forward(closedTS)
 }
 
-func proposeBatch(raftGroup proposerRaft, replID roachpb.ReplicaID, ents []raftpb.Entry) error {
+var everyErrProposalDropped = log.Every(time.Second)
+
+func proposeBatch(
+	ctx context.Context,
+	raftGroup proposerRaft,
+	replID roachpb.ReplicaID,
+	ents []raftpb.Entry,
+	props []*ProposalData, // must match ents slice
+) error {
+	if len(ents) != len(props) {
+		return errors.AssertionFailedf("ents and props don't match up: %v and %v", ents, props)
```

Comment:
Maybe - but it's a clear programming error, so I don't want to go crazy and definitely don't want to under-report information, in case the error is rare. I'll leave as is unless you feel strongly.

Reply:
I don't know how far up the stack this error can make, whether it can make into text logs or get outside the process. It can contain user data, right? So that's a bit sensitive. Maybe err on the safe side, and include only necessary info in the error message. See a similar `panic` in `maybeDeductFlowTokens`.